website/source/docs/providers/azurerm/r/loadbalancer_rule.html.markdown

```diff
@@ -0,0 +1,75 @@
+---
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_lb_rule"
+sidebar_current: "docs-azurerm-resource-loadbalancer-rule"
+description: |-
+  Create a LoadBalancer Rule.
+---
+
+# azurerm\_lb\_rule
+
+Create a LoadBalancer Rule.
+
+~> **NOTE When using this resource, the LoadBalancer needs to have a FrontEnd IP Configuration Attached
+
+## Example Usage
+
+```
+resource "azurerm_resource_group" "test" {
+    name = "LoadBalancerRG"
+    location = "West US"
+}
+
+resource "azurerm_public_ip" "test" {
+    name = "PublicIPForLB"
+    location = "West US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    public_ip_address_allocation = "static"
+}
+
+resource "azurerm_lb" "test" {
+    name = "TestLoadBalancer"
+    location = "West US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+
+    frontend_ip_configuration {
+      name = "PublicIPAddress"
+      public_ip_address_id = "${azurerm_public_ip.test.id}"
+    }
+}
+
+resource "azurerm_lb_rule" "test" {
+  location = "West US"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  loadbalancer_id = "${azurerm_lb.test.id}"
+  name = "LB Rule"
```

Comment:
The example name for azurerm_lb_rule given here is invalid (whitespace is an invalid character).
Consider a validate function for azurerm_lb_rule.name in the schema. The validation rules are in the error output below.

It would also be nice if name was the first attribute listed of the resource, rather than in the middle.

`* azurerm_lb_rule.hmrc: Error Creating/Updating LoadBalancer network.LoadBalancersClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="InvalidResourceName" Message="Resource name LB Rule is invalid. The name can be up to 80 characters long. It must begin with a word character, and it must end with a word character or with '_'. The name may contain word characters or '.', '-', '_'." Details=[]`